### ISSUE TITLE
[ENH]: Ability to Swap autopct and label positions

### ISSUE THREAD

**Issue by jet082** [matplotlib/matplotlib#24789]
### Problem

It would be nice to be able to have the labels applied on top of pie chart slices, rather than on the edges. Likewise, it would be nice to have the percentages on the outside of the pie chart slices instead of on the inside.

### Proposed solution

Allow an option to swap the autopct and label positions. Also, allow the autopct position to be autorotated via a flag like rotateautopct.

**Comment by vitorfrois** [comment]
Hello, I'd like to contribute on this issue.

**Comment by vitorfrois** [comment]
Actually it has a very simple solution.
At `def pie(...):`, a `swaplabelpct=False` parameter was added. Inside the function, I have the statement
`if swaplabelpct:
            labeldistance, pctdistance = pctdistance, labeldistance+0.05`
(the +0.05 allows some space between %s and the chart)
And when I print a pie chat with `swaplabelpct=True`, I get the following result:
![Figure_1](https://user-images.githubusercontent.com/46361092/208758991-703fad3b-76bc-4624-8428-625e4553cb8f.png)

**Comment by jklymak** [comment]
I'm not in favour of this request: `labeldistance` and `pctdistance` are both provided as kwargs.  The user can swap these themselves, or put in whatever value they prefer, if they do not like the default values.

**Comment by jklymak** [comment]
You can make it more explicit if you like, but the documentation is readily available - not _everything_ needs to be spelled out explicitly in an example.

**Comment by oscargus** [comment]
Are `pctdistance` and `labeldistance` using the same ""units"? Not so clear from the descriptions in that case....

**Comment by jklymak** [comment]
They are the same, but they should be spelled the same here.

**Comment by timhoffm** [comment]
> I'd honestly just move them next to each other in the docs but can't in this case 'cause of the signature

Just do it. Reorder according to the signature and then move *shadow* one place. Nobody will notice.

Additionally, we can start a kwarg only deprecation at *explode* to have the ability of reordering the signature in the future.

**Comment by oscargus** [comment]
> They are the same, but they should be spelled the same here.

OK! Then I think that spelling them the same way (not clear which, I may be inclined to think that radial distance is a bit clearer, although that is also expressed as a ratio, so the unit of the value is not so clear) will take this a long way.

**Comment by oscargus** [comment]
I'm a bit against not having the parameters in the "correct" order, but on the other hand pro making most methods with gazillion arguments keyword only after the most obvious ones (like data being positional, styling kw-only).

(btw, the current doc-string is already out of order with `normalize` before `labeldistance`)

**Comment by timhoffm** [comment]
> I'm a bit against not having the parameters in the "correct" order,

The rough order should be the same, but I think nobody will be confused or complain if the signature is

```
pctdistance=0.6, shadow=False, labeldistance=1.1,
```
but the documentation is

```
pctdistance
labeldistance
shadow
```
Personally, I mostly ignore the parameters in the signature because the parameter list is much more readable: https://matplotlib.org/stable/api/_as_gen/matplotlib.axes.Axes.pie.html#matplotlib.axes.Axes.pie
I only cross-check in very special cases.

**Comment by timhoffm** [comment]
> I also have this question about the sanctity of the signature and matching things up w.r.t. https://github.com/matplotlib/matplotlib/pull/24470 b/c I currently put hatch next to color and really don't want it at the bottom.

My take on this:
- Signature is API. Reordering can only be done after making parameters keyword-only.
- Generally, the order of the documented parameters should follow the signature.
- Slight reordering of documented parameters is ok if we think that logical parameter grouping will be be more beneficial to the reader than the exact order-matching to the signature. If we deviate, we should start a keyword-only deprecation to be able to adjust the signature parameter order later.

**Comment by oscargus** [comment]
I think that this can be closed now.

There is an example explicitly showing this: https://matplotlib.org/devdocs/gallery/pie_and_polar_charts/pie_features.html#swap-label-and-autopct-text-positions

and the documentation is updated:
https://matplotlib.org/devdocs/api/_as_gen/matplotlib.axes.Axes.pie.html#matplotlib.axes.Axes.pie
